src/encoder.rs

```diff
@@ -128,6 +128,24 @@ impl<W: Write> Writer<W> {
             return Err(EncodingError::Format("Zero height not allowed".into()));
         }
 
+        let bit_depth = self.info.bit_depth;
+        let color_type = self.info.color_type;
+
+        if (bit_depth == BitDepth::One || bit_depth == BitDepth::Two || bit_depth == BitDepth::Four)
+            && (color_type == ColorType::RGB
+                || color_type == ColorType::GrayscaleAlpha
+                || color_type == ColorType::RGBA)
+            || bit_depth == BitDepth::Sixteen && color_type == ColorType::Indexed
```

Comment:
I believe this can be written more clearly and separated into small helper function. And be it just in case we want to offer disabling the check.

Reply:
And a comment referring to the relevant section would be nice (via number, not link) to not confused readers why certina combinations are 'arbitrarily' disallowed.

Reply:
Moved to a helper function now. Good point on the comment.